mmf/datasets/processors/text_image_processors.py

```diff
@@ -0,0 +1,157 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+from math import nan
+
+import numpy as np
+import torch
+from mmf.common.registry import registry
+from mmf.datasets.processors.processors import BaseProcessor
+from mmf.utils.process_text_image import text_token_overlap_with_bbox
+from mmf.utils.text import is_punctuation
+from scipy import interpolate
+from transformers.tokenization_auto import AutoTokenizer
+
+
+class TextImageProcessor(BaseProcessor):
+    def __init__(self, config, *args, **kwargs):
+        super().__init__(config, *args, **kwargs)
+        self.config = config
+        text_tokenizer_config = config.text_processor.tokenizer_config
+        self.text_tokenizer = AutoTokenizer.from_pretrained(
+            text_tokenizer_config.type, **text_tokenizer_config.params
+        )
+
+    def sample_traces(self, utterances, utterance_times, traces):
+        tokens, index_tokens, token_weights, token_traces = [], [], [], []
+        for idx, utterance in enumerate(utterances):
+            if self.config.remove_punctuations:
+                utterance = self.remove_punctuation(utterance)
+
+            tokens_from_utterance = self.text_tokenizer.tokenize(utterance)
+            indexed_tokens_from_utterance = self.text_tokenizer.convert_tokens_to_ids(
+                tokens_from_utterance
+            )
+            (
+                token_times,
+                token_weights_from_utterance,
+            ) = self.determine_token_weight_and_times(
+                tokens_from_utterance,
+                utterance_times[idx][0].item(),
+                utterance_times[idx][1].item(),
+            )
+            token_traces_from_utterance = self.traces_from_token(
+                token_times, traces, self.config.trace_num_samples
+            )
+
+            tokens += tokens_from_utterance
+            index_tokens += indexed_tokens_from_utterance
+            token_weights += token_weights_from_utterance
+            token_traces += token_traces_from_utterance
+        return (
+            tokens,
+            torch.IntTensor(index_tokens),
+            torch.FloatTensor(token_weights),
+            torch.FloatTensor(token_traces),
+        )
+
+    def determine_token_weight_and_times(self, tokens, start, end):
+        return (
+            torch.FloatTensor([start, end]).repeat(len(tokens), 1),
+            torch.FloatTensor([1 / len(tokens)]).repeat(len(tokens)),
```

Comment:
In general, use `torch.tensor(x, dtype=torch.float)` etc for creating tensors of type `float`. Similarly for other types as well.